internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/hashicorp/terraform-ls/internal/job"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	var ids job.IDs
+
+	for _, change := range params.Changes {
+		uri := string(change.URI)
+		_, err := os.Stat(uri)
+		if err != nil {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("Unable to update: %s, Failed to open directory", uri),
+			})
```

Comment:
I'm not sure we should be opening message windows for errors here. This pops for any change to any file/folder, so it could open many message windows at once. What is the user supposed to do if there is an error updating our cache? How will sending this error help the user?

Thinking about this, if we can't stat a directory we can't 'watch' it, therefore there is nothing for us to do. Likewise for a file. I would rather log here than send messages the user can't do anything about.

This also ties into the larger debate we have about returning errors from these handlers and how that affects client experience, but that isn't something you are required to deal with here

Reply:
I think the main point here is that clients shouldn't be sending unsupported files, so this indicates client-side bug, which we'd like the user to report. We can make that clearer in the message though.

Reply:
That's a very valid point. I was also thinking about how it can quickly become unwanted noise for the user. Logging makes sense here, especially since this is supposed to be an implicit request to the server unlike other events such as "go to definition", in which case, telling the user that the particular file isn't supported makes a lot more sense.